tools/src/main/scala/scala/scalanative/compiler/codegen/GenTextualLLVM.scala

```diff
@@ -84,9 +84,36 @@ class GenTextualLLVM(assembly: Seq[Defn]) extends GenShow(assembly) {
 
     val isDecl  = blocks.isEmpty
     val keyword = if (isDecl) "declare" else "define"
-    val params =
-      if (isDecl) r(argtys, sep = ", ")
-      else r(blocks.head.params: Seq[Val], sep = ", ")
+
+    def showDefnArg(arg: Arg,
+                    value: Val.Local): (Show.Result, Seq[Show.Result]) =
+      arg match {
+        case Arg(_, ArgAttrs.empty) => (sh"${value: Val}", Seq.empty)
+        case Arg(Type.Ptr, ArgAttrs(Some(pointee), None)) =>
+          val pointer = fresh()
+          (sh"$pointee* byval %$pointer",
+           Seq(sh"%${value.name} = bitcast $pointee* %$pointer to i8*"))
+        case Arg(Type.Ptr, ArgAttrs(None, Some(pointee))) =>
+          val pointer = fresh()
+          (sh"$pointee* sret %$pointer",
+           Seq(sh"%${value.name} = bitcast $pointee* %$pointer to i8*"))
+        case x => unsupported(x)
```

Comment:
Given that only a single annotation may be present, it might make sense to have a single field in the `Arg` class of type `PassConv` (i.e. passing convention) that has `sret` and `byval` variants. It will also make it impossible to accidentally have two at the same time by construction.